api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -532,7 +533,14 @@ public void setVisit(Visit visit) {
 		
 		return encounterProviders.stream()
 				.filter(ep -> includeVoided || !ep.getVoided())
-				.collect(Collectors.groupingBy(EncounterProvider::getEncounterRole, Collectors.mapping(EncounterProvider::getProvider, Collectors.toSet())));
+				.collect(Collectors.toMap((ep) -> (ep).getEncounterRole(),
```

Comment:
I wasn't aware role was optional, sorry! I also agree with Ivo, it's a bit hard to read. Can you please revert to the previous implementation without streams? I'd do it but I don't have permissions.